addons/rating/models/mail_message.py

```diff
@@ -7,51 +7,42 @@
 class MailMessage(models.Model):
     _inherit = 'mail.message'
 
-    rating_ids = fields.One2many('rating.rating', 'message_id', groups='base.group_user', string='Related ratings')
+    rating_ids = fields.One2many("rating.rating", "message_id", string="Related ratings")
+    rating_id = fields.Many2one("rating.rating", compute="_compute_rating_id")
     rating_value = fields.Float(
         'Rating Value', compute='_compute_rating_value', compute_sudo=True,
         store=False, search='_search_rating_value')
 
+    @api.depends("rating_ids.consumed")
+    def _compute_rating_id(self):
+        for message in self:
+            message.rating_id = message.rating_ids.filtered(lambda rating: rating.consumed).sorted(
+                "create_date", reverse=True
+            )[:1]
+
     @api.depends('rating_ids', 'rating_ids.rating')
     def _compute_rating_value(self):
-        ratings = self.env['rating.rating'].search([('message_id', 'in', self.ids), ('consumed', '=', True)], order='create_date DESC')
-        mapping = dict((r.message_id.id, r.rating) for r in ratings)
         for message in self:
-            message.rating_value = mapping.get(message.id, 0.0)
+            message.rating_value = message.rating_id.rating if message.rating_id else 0.0
 
     def _search_rating_value(self, operator, operand):
         ratings = self.env['rating.rating'].sudo().search([
             ('rating', operator, operand),
-            ('message_id', '!=', False)
+            ('message_id', '!=', False),
+            ("consumed", "=", True),
```

Comment:
Note: the search was not the exact match of the compute due to missing `consumed`.